### DISCUSSION TITLE
How to return the Wizard to step 1 after create success in custom page.

### DISCUSSION THREAD

**Discussion by DevYSM** [filamentphp/filament#12526] [Help] [answered]
### Package

Form builder

### Package Version

v3

### How can we help you?

Hello, I have created a custom page using the form following schema this form saves the data into external API.
```php
Wizard::make([
    Wizard\Step::make('Order')
        ->schema([
            // ...
        ]),
    Wizard\Step::make('Delivery')
        ->schema([
            // ...
        ]),
    Wizard\Step::make('Billing')
        ->schema([
            // ...
        ]),
])
```
Everything is good, But after the creation successfully I'm using $this->form->fill() this method clears the data but doesn't reset the wizard to the first step.

Any Idea?

Best Regards.

**Comment by DevYSM** [comment]
@leandrocfe

**Reply by leandrocfe** [reply, marked as answer]
Check this

https://www.youtube.com/live/z5StCXCT7mQ?si=yDA9s0qfWurVXbLW

~58:13

**Comment by DevYSM** [comment]
I have found the solution, Just add this line to the 
```php
Wizard::make()->extraAlpineAttributes([
    '@form-submitted.window' => "step='main_details'" // pass the wizard name here.
]

// And add the following line to the submit or create method

```

```php
  $this->dispatch('form-submitted');
```

@leandrocfe Thank you very much :heart:

**Comment by Martins-74** [comment]
For Filament V4 add the following to the Wizard, this makes sure that the wizard resets always to the first step even if the step name gets changed.
`
Wizard::make()->extraAlpineAttributes([
                        '@form-submitted.window' => "step=getSteps()[0]"
   ])
`

and then just call the event the same way @DevYSM is doing
` 
$this->dispatch('form-submitted');
`
